src/operations/common_functions.ts

```diff
@@ -43,26 +43,37 @@ export async function indexInformation(
   return info;
 }
 
-export function prepareDocs(
+export function maybeAddIdToDocuments(
   coll: Collection,
   docs: Document[],
   options: { forceServerObjectId?: boolean }
-): Document[] {
+): Document[];
+export function maybeAddIdToDocuments(
+  coll: Collection,
+  docs: Document,
+  options: { forceServerObjectId?: boolean }
+): Document;
+export function maybeAddIdToDocuments(
+  coll: Collection,
```

Comment:
Just curious, not at all essential, if this could be made type-safe since we have a reference to the collection, could TSchema be threaded through along with InferId 🤔 is pkFactory parameterized on TSchema?

Reply:
Not essential but since you're "here". You could delete that TODO, js-bson is typed but I don't think it would ever be helpful to constrain the return type 🤔 at least not to a BSON type, maybe to the TSchema but that's a thought for another time.